### PR TITLE
Reuse a TimelineTask for the scheduler frame and animate events

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/flutter/issues/90061

### CODE REVIEW

File: packages/flutter/lib/src/scheduler/binding.dart [hunk 1]
```diff
@@ -996,8 +996,7 @@ mixin SchedulerBinding on BindingBase {
     handleDrawFrame();
   }
 
-  TimelineTask? _frameTimelineTask;
-  TimelineTask? _animateTimelineTask;
+  TimelineTask _frameTimelineTask = TimelineTask();
```
Comment (goderbauer) [inline packages/flutter/lib/src/scheduler/binding.dart:999]: final?

**Review by goderbauer — APPROVED** [review]
LGTM once checks are happy

File: packages/flutter/lib/src/scheduler/binding.dart (hunk 1 above)
Comment (jason-simmons) [reply packages/flutter/lib/src/scheduler/binding.dart]: done

**Comment by jason-simmons** [comment]
Updated this to skip logging these events in release mode.

See https://github.com/flutter/flutter/issues/90061#issuecomment-921333373

**Pull request merged**